### PR TITLE
[Tchap-v2] Setting up the MCS

### PR DESCRIPTION
**Pull request by Yoshiin**
Setting up the MCS.

- #39 
- #70 
- #97 
- #77 
- #216 

See #201 for more informations.

### CODE REVIEW

File: src/utils/ContentScanner.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+     * @param isThumb If the requested data will be a thumbnail.
+     * @returns {*} A string or an error object.
+     */
+    static downloadContent(content, isThumb = false) {
+        const baseUrl = MatrixClientPeg.get()['baseUrl'];
+
+        if (content.url !== undefined) {
+            const fileUrl = content.url.split('//')[1];
+            let url;
+            if (isThumb) {
```
Comment (giomfo) [inline src/utils/ContentScanner.js]: When the boolean value `isThumb` is true, you must consider first `content.info.thumbnail_url` (if any).
We can use the url built by adding `thumbnailParams` **only** for attached image, when no `thumbnail_url` is provided.

For example, see attached video `thumbnail_url`  here: https://github.com/dinsic-pim/matrix-react-sdk/pull/218/files#diff-22ec6bbff59ebc5c6e87770509f52416R85)

File: src/utils/ContentScanner.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                    console.error(err);
+                    return this.generateError(false, "Error: Cannot fetch the file");
+                }));
+        } else {
+            return this.generateError(false, 'Error: This is not a matrix content');
+        }
+    }
+
+    /**
+     * Download an unencrypted content.
```
Comment (giomfo) [inline src/utils/ContentScanner.js]: The comment is wrong here, this method does not download the content

File: src/utils/ContentScanner.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    /**
+     * Download an encrypted content.
+     * @param content A Mtrix Event content.
+     * @param isThumb If the requested data will be a thumbnail.
+     * @returns {Promise<*>} A Promise or an error object.
+     */
+    static async downloadContentEncrypted(content, isThumb = false) {
+        if (content.file !== undefined || content.info.thumbnail_file !== undefined) {
+            let file = isThumb ? content.info.thumbnail_file : content.file;
```
Comment (giomfo) [inline src/utils/ContentScanner.js]: You should set `file` first, then check whether it is defined, no?

File: src/utils/ContentScanner.js [hunk 4]
```diff
@@ -82,20 +86,26 @@ class ContentScanner {
...
         const baseUrl = MatrixClientPeg.get()['baseUrl'];
+        let url;
 
-        if (content.url !== undefined) {
-            const fileUrl = content.url.split('//')[1];
-            let url;
+        if (content.url !== undefined || content.info.thumbnail_url !== undefined) {
             if (isThumb) {
-                url = `${baseUrl + TchapApi.downloadUnencryptedThumnailUrl}${fileUrl}${TchapApi.thumbnailParams}`;
+                if (content.info.thumbnail_url) {
```
[the remark this answers is not in the document]
Comment (Yoshiin) [reply src/utils/ContentScanner.js]: Indeed

File: src/utils/ContentScanner.js [hunk 5]
```diff
@@ -106,21 +116,29 @@ class ContentScanner {
...
+     * @param {object} content A Mtrix Event content.
+     * @param {boolean} isThumb If the requested data will be a thumbnail.
+     * @returns {Promise<*>|blob} A Promise or an error object.
      */
-    static async downloadContentEncrypted(content, isThumb = false) {
-        if (content.file !== undefined || content.info.thumbnail_file !== undefined) {
-            let file = isThumb ? content.info.thumbnail_file : content.file;
-            let blob = await decryptFile(file);
+    static async downloadEncryptedContent(content, isThumb = false) {
+        // check if is thumb prev
```
[the remark this answers is not in the document]
Comment (Yoshiin) [reply src/utils/ContentScanner.js]: I forgot to remove it.

**Pull request merged**